src/ai/ai-brain.ts

```diff
@@ -116,24 +129,27 @@ export class AiBrain {
         return closestPlayer;
     }
 
-    private targetClosestPlayer(myPlayer: Player, closestPlayer: Player) {
+    private targetObject(myPlayer: Player, object: GameObject) {
         const direction = Vector.diff(
             myPlayer.getCollider().getPosition(),
             // aim a bit in front of the target
-            closestPlayer.getCollider().getPosition().getSum(closestPlayer.getForward().getScaled(0.5)),
+            object.getCollider().getPosition().getSum(object.getForward().getScaled(0.5)),
         );
         this.targetAngle = direction.toAngle();
     }
 
     private isCollisionImminent(myPlayer: Player, context: GameContext) {
         let result = false;
+        const COLLISION_CRITICAL_TIME = 2; // seconds
         context.obstacles.forEach((o) => {
-            result = result || isCrashImminent(
+            const t = getTimeBeforeCollision(
                 myPlayer.getCollider(),
                 o.getCollider(),
                 myPlayer.getForward(),
-                o.getForward(),
-                2);
+                o.getForward());
+
+            if (t !== undefined)
+                result = result || t < COLLISION_CRITICAL_TIME; 
```

Comment:
This can be simplified to
```ts
return t < COLLISION_CRITICAL_TIME;
```
if `t` is `null`, it will compare `false` with anything, if it is number, it will return true for values less than collision time.